### DISCUSSION TITLE
Set the base path configuration option using an environment variable

### DISCUSSION THREAD

**Discussion by HCl-not-HCi** [withastro/roadmap#248] [Proposal]
There is a `base` option in the `astro.config.mjs` file to configure the base path of the website (e.g. `/my-path/` to deploy under `example.com/my-path/`). In my case, I had to deploy the page in different environments with different paths. Meaning, I have to create a dedicated config file for each path.

First, I expected the possibility to configure the path by using an environment variable (since the base URL gets exported to the `import.meta.env.BASE_URL` env variable). However, this is not possible.

Therefore, I wish there would be the possiblity to **configure the `base` config option using an environment variable**. This could be extended to support other options as well (where it makes sense) to be fully flexible when buidling and deploying a website.

In already started a discussion about this on Discord, but we did not come to a solution:
https://discord.com/channels/830184174198718474/845451724738265138/998514489710358568

**Comment by BigglesZX** [comment]
For anyone looking for a resolution to this question, it seems to be possible now and is documented [here](https://docs.astro.build/en/guides/environment-variables/#in-the-astro-config-file).
